src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -539,6 +540,43 @@ private void processConverterStation(LfNetwork lfNetwork, HvdcConverterStation<?
 
     }
 
+    private Map<Generator, GeneratorState> applyGeneratorContingency(Network network, PropagatedContingency contingency) {
+        // It applies on the network the loss of the generators contained in the contingency.
+        // Generator states are stored.
+        Collection<Generator> generatorsToSave = new HashSet<>();
+        for (String generatorId : contingency.getGeneratorIdsToLose()) {
+            Generator generator = network.getGenerator(generatorId);
+            generatorsToSave.add(generator);
+        }
+        Map<Generator, GeneratorState> generatorStates = GeneratorState.createGeneratorStates(generatorsToSave);
+        for (String generatorId : contingency.getGeneratorIdsToLose()) {
+            Generator generator = network.getGenerator(generatorId);
+            generator.setTargetP(0);
+            generator.setTargetQ(0);
+            generator.setVoltageRegulatorOn(false);
+        }
+
+        return generatorStates;
+    }
+
+    private List<ParticipatingElement> removeGeneratorParticipation(Network network, LfNetwork lfNetwork, PropagatedContingency contingency, List<ParticipatingElement> participatingElements) {
```

Comment:
I think you don't need a dedicated function for that. It could be done with the other attributes of the generator.